File: backend/src/main/kotlin/com/millo/hayoungplace/party/domain/Party.kt
package com.millo.hayoungplace.party.domain

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.time.LocalDateTime

@Document(collection = "parties")
data class Party(
    @Id
    val id: String? = null,
    val title: String,
    val status: PartyStatus = PartyStatus.RECRUITING,
    val description: String,
    val location: String,
    val date: LocalDateTime,
    val maxMembers: Int? = null,
    val tags: List<String> = emptyList(),
    val nickname: String,
    val password: String,
    val createdAt: LocalDateTime = LocalDateTime.now(),
    val updatedAt: LocalDateTime = LocalDateTime.now()
)

File: frontend/src/types/party.ts
export enum PartyStatus {
  RECRUITING = 'RECRUITING',
  COMPLETED = 'COMPLETED'
}

export interface Party {
  id?: string
  title: string
  status: PartyStatus
  description: string
  location: string
  date: string // ISO string
  maxMembers?: number
  currentMembers: number
  tags: string[]
  nickname: string
  createdAt: string
  dDay?: number
}

export interface PartyCreateRequest {
  title: string
  description: string
  location: string
  date: string // ISO string
  maxMembers: number
  tags: string[]
  nickname: string
  password: string
}

export interface PartyUpdateRequest {
  title: string
  description: string
  location: string
  date: string // ISO string
  maxMembers?: number
  tags: string[]
  status?: PartyStatus
}

export interface PartyMember {
  id?: string
  nickname: string
  isCreator: boolean
  joinedAt: string
}

export interface PartyMemberRequest {
  nickname: string
  password: string
}

export interface PartyListResponse {
  content: Party[]
  totalElements: number
  totalPages: number
  size: number
  number: number
  numberOfElements: number
  first: boolean
  last: boolean
} 